### PR TITLE
Adds features for admin user for eagle

### PR DESCRIPTION
**Pull request by utkarsh867**

### CODE REVIEW

**Comment by utkarsh867** [comment]
Will merge once all features are implemented

File: prisma/datamodel.prisma [hunk 1]
```diff
@@ -11,6 +11,12 @@ type User {
     sessions: [UserSessions] @relation(name: "UserLoginSessions", onDelete: CASCADE)
 }
 
+type Admin {
+    id: ID! @id
+    username: String! @unique
+    password: String!
```
Comment (98sean98) [inline prisma/datamodel.prisma]: should we even return the encrypted password when an `Admin` is queried?

File: src/query/user.ts [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
     username,
   });
 };
 
+export const adminUser = (parent, { username }, ctx): Promise<Admin> => {
+  return ctx.prisma.admin({
+    username,
+  });
+};
+
```
Comment (98sean98) [inline src/query/user.ts]: I think this would return the encrypted password. Its fine, but not recommended right?

Comment (utkarsh867) [reply src/query/user.ts]: yes, we don't want to add the password field, even if it is encrypted

Comment (98sean98) [reply src/query/user.ts]: Do you want me to help resolve this with a new `type Credential` linked to `type Admin`?

Comment (utkarsh867) [reply src/query/user.ts]: Let's take this to today's meeting too

Comment (98sean98) [reply src/query/user.ts]: I'll work on this

**Comment by 98sean98** [comment]
@utkarsh867 please review changes and merge

**Comment by utkarsh867** [comment]
@98sean98 This should take a very different approach because of the recent developments in how we handle login etc.

**Comment by 98sean98** [comment]
I see. We can drop #44 in favour of this instead. I'll spend a bit of time to go through the code base in detail for this.

**Pull request closed without merging**